src/segment-loader.js

```diff
@@ -34,27 +34,27 @@ import {lastBufferedEnd} from './ranges.js';
  * @param {Object} playlist - the playlist object to look for a
  * @return {number} An index of a segment from the playlist to load
  */
-export const getSyncSegmentCandidate = function(currentTimeline, {segments = []} = {}) {
-  // if we don't currently have a real timeline yet.
-  if (currentTimeline === -1) {
-    return 0;
-  }
+export const getSyncSegmentCandidate = function(currentTimeline, playlist, targetTime) {
+  const segments = playlist.segments || [];
 
-  const segmentIndexArray = segments.reduce((acc, s, i) => {
-    if (s.timeline === currentTimeline) {
-      acc.push(i);
-    }
-    return acc;
-  }, []);
+  const timelineSegments = [];
+  let time = 0;
+
+  for (let i = 0; i < segments.length; i++) {
+    const segment = segments[i];
+
+    if (currentTimeline === segment.timeline) {
+      timelineSegments.push(i);
+      time += segment.duration;
 
-  if (segmentIndexArray.length) {
-    // TODO: why do we do this? Basically we choose index 0 if
-    // segmentIndexArray.length is 1 and index = 1 if segmentIndexArray.length
-    // is greater then 1
-    return segmentIndexArray[Math.min(segmentIndexArray.length - 1, 1)];
+      if (time > targetTime) {
+        return i;
+      }
+    }
   }
 
-  return Math.max(segments.length - 1, 0);
+  // default to grabbing the last timeline segment or zero.
+  return Math.max(0, timelineSegments[timelineSegments.length - 1] || 0);
```

Comment:
question: so, because of how `timelineSegments` are generated (based on whether they're in the current timeline or not), when we get here we either have timelineSegments or we have an empty list and so we end up returning `0` as a default like we did previously when `currentTimeline === -1`?

Reply:
yes thats correct.